pyodide-build/pyodide_build/io.py

```diff
@@ -93,6 +93,7 @@ class _BuildSpec(BaseModel):
     script: str | None = None
     post: str | None = None
     unvendor_tests: bool = Field(True, alias="unvendor-tests")
+    retain_test_patterns: list[str] = Field([], alias="retain-test-patterns")
```

Comment:
I think there's room for more generalization for this, but I don't want to block this PR for that. So, how about adding an underscore prefix indicate that it's private? (Related: https://github.com/pyodide/pyodide/issues/3092)